pkg/asset-manager-utils/test/SinglePoolAaveATokenAssetManager.test.ts

```diff
@@ -0,0 +1,330 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+
+import { bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import { advanceTime } from '@balancer-labs/v2-helpers/src/time';
+import { encodeJoinWeightedPool } from '@balancer-labs/v2-helpers/src/models/pools/weighted/encoding';
+
+const OVER_INVESTMENT_REVERT_REASON = 'investment amount exceeds target';
+const UNDER_INVESTMENT_REVERT_REASON = 'withdrawal leaves insufficient balance invested';
+
+const tokenInitialBalance = bn(200e18);
+const amount = bn(100e18);
+
+const setup = async () => {
+  const [, admin, lp, other] = await ethers.getSigners();
+
+  const tokens = await TokenList.create(['DAI', 'MKR'], { sorted: true });
+
+  // Deploy Balancer Vault
+  const vaultHelper = await Vault.create({ admin });
+  const vault = vaultHelper.instance;
+
+  // Deploy mocked Aave
+  const lendingPool = await deploy('MockAaveLendingPool', { args: [] });
+  const aaveRewardsController = await deploy('MockAaveRewards');
+  const stkAave = aaveRewardsController;
+
+  const daiAToken = await deploy('MockAToken', { args: [lendingPool.address, 'aDai', 'aDai', 18] });
+  await lendingPool.registerAToken(tokens.DAI.address, daiAToken.address);
+
+  // Deploy Asset manager
+  const assetManager = await deploy('SinglePoolAaveATokenAssetManager', {
+    args: [
+      vault.address,
+      tokens.DAI.address,
+      lendingPool.address,
+      daiAToken.address,
+      aaveRewardsController.address,
+      stkAave.address,
```

Comment:
In the real system these two are not the same contract, right? Meaning stkAave is standalone, separate from the rewards controller.

Reply:
correct